File: C++/DataStructures/linkedlist.cpp
/* Linked List Implementation in C++ */
#include <iostream>
using namespace std;

class Node{
    public:
        int data;
        Node *next = NULL;
};

class List : public Node{

    private:
        Node *head;
        int size = 0;

    
    public:

        // Constructor
        List(){
            head = NULL;
        }

        // Add a node
        void addNode(int value){
            Node *newNode = new Node();
            Node *tempNode = new Node();
            newNode->data = value;

            if(head == NULL){
                head = newNode;
            }
            else{
                tempNode = head;
                while(tempNode->next != NULL){
                    tempNode = tempNode->next;
                }
                tempNode->next = newNode;
            }
            size++;
        }

        // Print the list
        void printList(){
            Node *tempNode;
            tempNode = head;
            while(tempNode->next != NULL){
                cout << tempNode->data << "--->";
                tempNode = tempNode->next;
            }
            cout << tempNode->data;
        }

        // Remove from last
        void remLast(){
            Node *temp;
            temp = head;

            while(temp->next != NULL){
                temp = temp->next;
            }
            free(temp);
            size--;
        }

        // Remove node from a given position
};

int main(){
    List *l1 = new List();
    l1->addNode(10);
    l1->addNode(20);
    l1->addNode(30);
    l1->printList();
    l1->remLast();
    cout << "----" << endl;
    l1->printList();
}

File: JavaScript/DataStructures/linkedlist.js
/* Linked List in javascript */

class Node{
    constructor(data , next = null ){
        this.data = data;
        this.next = next;
    }
}

class LinkedList{
    constructor(){
        this.size = 0;
        this.head = null;
    }

    // Print Size
    printSize(){
        console.log(this.size);
    }
    
    // Print the complete list
    printList(){
        let current = this.head;
        while(current){
            console.log(current.data);
            current = current.next;
        }
    }
    
    // Insert at last
    insertLast(value){
        let node = new Node(value);

        if(!this.head){
            this.head = node;
            this.size++;
        }
        else{
            let current = this.head;
            while(current.next){
                current = current.next;
            }         
            current.next = node;
            this.size++;
        }
    }

    // Insert at beginning 
    insertFirst(value){
      this.head = new Node(value , this.head);
        this.size++;
    }
}

ll = new LinkedList();
ll.insertLast(10);
ll.insertLast(20);
ll.insertLast(30);
ll.insertLast(40);
ll.printList();
ll.printSize();